src/app.js

```diff
@@ -1,20 +1,64 @@
-const express = require('express');
-const app = express();
+require('dotenv').config()
+const express = require('express')
+const app = express()
+const morgan = require('morgan')
+const bodyParser = require('body-parser')
 const cors = require('cors')
-const favicon = require('express-favicon');
-const logger = require('morgan');
+const favicon = require('express-favicon')
+const mainRouter = require('./routes/mainRouter.js')
+const helmet = require('helmet')
+const session = require('express-session')
+const passport = require('passport')
+const mongoose = require('mongoose')
+const asyncErrors = require('express-async-errors')
+const cookieParser = require('cookie-parser')
+const rateLimit = require('express-rate-limit')
+const xss = require('xss')
+const mongoSanitize = require('mongo-sanitize')
+const limiter = rateLimit({
+  windowMs: 15 * 60 * 1000, // 15 minutes
+  max: 60 // limit each IP to 100 requests per windowMs
+})
 
-const mainRouter = require('./routes/mainRouter.js');
-
-// middleware
-app.use(cors());
-app.use(express.json());
-app.use(express.urlencoded({ extended: false }));
-app.use(logger('dev'));
+// middleware setup
+app.use(cors())
+app.use(xss())
+app.use(limiter)
+app.use(mongoSanitize())
+app.use(express.json())
+app.use(helmet())
+app.use(express.urlencoded({ extended: false }))
+app.use(morgan('dev'))
 app.use(express.static('public'))
-app.use(favicon(__dirname + '/public/favicon.ico'));
+app.use(favicon(__dirname + '/public/favicon.ico'))
+app.use(bodyParser.json())
+app.use(passport.initialize())
+app.use(passport.session())
+app.use(cookieParser())
+app.use(
+  session({ secret: 'your-secret-key', resave: false, saveUninitialized: true })
+)
```

Comment:
Based on the docs, these two Passport lines should be defined _after_ you define the Express Session middleware, since Passport will be adding data to that: https://www.npmjs.com/package/passport#:~:text=%7D)%3B-,Middleware,-To%20use%20Passport

```suggestion
app.use(cookieParser())
app.use(
  session({ secret: 'your-secret-key', resave: false, saveUninitialized: true })
)
app.use(passport.initialize())
app.use(passport.session())
```

Reply:
it is corrected , commit- 53547538447ea06d0d06645414122fc910ef604e